osu.Game/Screens/Play/HUD/LegacyComboSplash.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Skinning;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public class LegacyComboSplash : CompositeDrawable, ISkinnableDrawable
+    {
+        public Bindable<int> Current { get; } = new BindableInt { MinValue = 0 };
+
+        public bool UsesFixedAnchor { get; set; }
+
+        private Random random = new Random();
+
+        public LegacyComboSplash()
+        {
+            AlwaysPresent = true;
+            AutoSizeAxes = Axes.Both;
+            Origin = Anchor.CentreLeft;
+            Current.BindValueChanged(e => OnNewCombo(e.NewValue), false);
+        }
+
+        private void OnNewCombo(int combo)
+        {
+            if (shouldDisplay(combo) && InternalChildren.Count != 0)
+            {
+                Drawable sprite = InternalChildren[random.Next(0, InternalChildren.Count)];
+                sprite.FadeTo(1).MoveToX(-sprite.Width).Then().MoveToX(0, 1000, Easing.OutCirc).FadeOut(1000, Easing.OutQuad);
```

Comment:
If I'm looking at stable source correctly this should be closer to
```suggestion
                sprite.MoveToX(-sprite.Width * 0.625f).Then().MoveToX(0, 700, Easing.Out);
                sprite.FadeTo(1).Delay(200).FadeOut(1000, Easing.In);
```
I've also split the X and opacity transforms into 2 lines because it was confusing reading as written.